play-services-location/core/provider/src/main/kotlin/org/microg/gms/location/network/wifi/MovingWifiHelper.kt

```diff
@@ -247,7 +247,9 @@ class MovingWifiHelper(private val context: Context) {
     
     private fun parseSncf(location: Location, data: ByteArray): Location {
         val json = JSONObject(data.decodeToString())
-        if(json.getInt("fix") == -1) throw RuntimeException("GPS not valid")
+        json.optInt("fix").takeIf { !it.isNaN() }?.let {
```

Comment:
Int can't be NaN. Also, if this field is just not present in Intercites, you can just change the condition to `json.has("fix") && json.getInt("fix") == -1`

Reply:
Thanks!